benchmarks/benchmarks.py

```diff
@@ -0,0 +1,124 @@
+import json
+import pathlib
+import os
+import pandas as pd
+import matplotlib.pyplot as plt
+import pytest
+import argparse
+import glob
+from pathlib import Path
+
+
+def unravel(data, key):
+    """Transforms {key:{another_key: values, another_key2: value2}} into
+    {key_another_key:value, key_another_key2:value}"""
+    for d in data:
+        values = d.pop(key)
+        for k, v in values.items():
+            d[key+'_'+k] = v
+    return data
+
+
+def benchmark_to_dataframe(filepath):
+    """Loads a JSON file where the benchmark is stored and returns a dataframe
+    with the benchmar information."""
+    path = pathlib.Path(__file__).absolute().parents[1].joinpath(filepath)
+    with open(path) as f:
+        data = json.load(f)
+        data = data['benchmarks']
+        data = unravel(data, 'options')
+        data = unravel(data, 'stats')
+        data = unravel(data, 'params')
+        data = unravel(data, 'extra_info')
+        data = pd.DataFrame(data)
+
+        # Set operation properly (for example: matmul instead of:
+        # UNSERIALIZABLE[<function Qobj.__mammal__ at 0x...)
+        # The name of the operation is obtained from the group name
+        data.params_get_operation = data.group.str.split('-')
+        data.params_get_operation = [d[-1] for d in data.params_get_operation]
+
+        return data
+
+
+def plot_benchmark(df, destination_folder):
+    """Plots results using matplotlib. It iterates params_get_operation and
+    params_density and plots time vs N (for NxN matrices)"""
+    grouped = df.groupby(['params_get_operation', 'params_density'])
+    for (operation, density), group in grouped:
+        for dtype, g in group.groupby('extra_info_dtype'):
+            plt.errorbar(g.params_size, g.stats_mean, g.stats_stddev,
+                         fmt='.-', label=dtype)
+
+        plt.title(f"{operation} {density}")
+        plt.legend()
+        plt.yscale('log')
+        plt.xscale('log')
+        plt.savefig(f".benchmarks/figures/{operation}_{density}.png")
+        plt.xlabel("N")
+        plt.ylabel("t(s)")
+        plt.close()
+
+
+def run_benchmarks(args):
+    "Run pytest benchmar with sensible defaults."
+    pytest.main(["--benchmark-only",
+                 "--benchmark-columns=Mean,StdDev,rounds,Iterations",
+                 "--benchmark-sort=name",
+                 "--benchmark-autosave",
+                 "-Wdefault"] +
+                args)
+
+
+def get_latest_benchmark_path():
+    """Returns the path to the latest benchmark run."""
+
+    benchmark_paths = glob.glob("./.benchmarks/*/*.json")
+    dates = [''.join(_b.split("/")[-1].split('_')[2:4])
+             for _b in benchmark_paths]
+    benchmarks = {date: value for date, value in zip(dates, benchmark_paths)}
+
+    dates.sort()
+    latest = dates[-1]
+    benchmark_latest = benchmarks[latest]
+
+    return benchmark_latest
+
+
+def main(args=[]):
+    parser = argparse.ArgumentParser()
```

Comment:
Could we perhaps add `description="...."` description of what the command does so that `--help` prints a one sentence of the script's function in addition to the parameters?

Reply:
Indeed, I already added one but forgot to push it. Should be included now. Thank you for spotting this!